### PR TITLE
When auto-importing the template, if the default 'Templates' folder d…

### PR DESCRIPTION
**Pull request by ptrivedi**
…oes not exist, create it

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/632#" title="Author self-approved">ptrivedi</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [ptrivedi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/executables/govc.go
Comment (abhinavmpandey08) [inline pkg/executables/govc.go:273]: `templateDir` already includes the `/` in the front, so we shouldn't need to add it here manually
https://github.com/aws/eks-anywhere/blob/3992ec57de953137f10ebcb8360dd2768d19e1db/pkg/providers/vsphere/vsphere.go#L859

Comment (ptrivedi) [reply pkg/executables/govc.go:273]: templateDir being passed into DeployTemplateFromLibrary is just the dir

https://github.com/aws/eks-anywhere/blob/3992ec57de953137f10ebcb8360dd2768d19e1db/pkg/executables/govc.go#L244

File: pkg/executables/govc.go
Comment (vivek-koppuru) [inline pkg/executables/govc.go:315]: So this would retry the folder.info call if folder.create fails right? Can we move them to separate retries, where we retry the folder.create outside of the folder.info retry?

Comment (ptrivedi) [reply pkg/executables/govc.go:315]: Oh good catch, yes, will move.

Comment (vivek-koppuru) [reply pkg/executables/govc.go:315]: Let's check for "not found" string here instead, it will clean up the logical flow

File: pkg/executables/govc.go
Comment (vivek-koppuru) [inline pkg/executables/govc.go:339]: Can we do this?
```suggestion
		})
	        if err != nil {
		         return fmt.Errorf("error creating folder: %v", err)
	         }
	}
```

Comment (vivek-koppuru) [reply pkg/executables/govc.go:339]: Actually, we are resetting err here by returning nil right?

**Review by vivek-koppuru — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Pull request merged**
